web/modules/custom/joinup_core/src/EventSubscriber/RefreshCachedFieldsEventSubscriber.php

```diff
@@ -86,89 +87,98 @@ public function __construct(EntityTypeManagerInterface $entityTypeManager, TimeI
   /**
    * {@inheritdoc}
    */
-  public function refreshExpiredField(RefreshExpiredFieldEventInterface $event) {
-    // Only react to fields that store the download count or visit count.
-    if (!in_array($event->getFieldName(), [static::FIELD_NAME_DOWNLOAD_COUNT, static::FIELD_NAME_VISIT_COUNT])) {
-      return;
+  public function refreshExpiredFields(RefreshExpiredFieldsEventInterface $event) {
+    $items = $event->getExpiredItems()->getItems();
+
+    // All requests are sent by POST method to handle the amount of concurrent
+    // requests in terms of request length.
+    $this->piwikQueryFactory->getQueryFactory()->getHttpClient()->setMethod('POST');
+    $query = $this->piwikQueryFactory->getQuery('API.getBulkRequest');
+
+    $piwik_config = $this->configFactory->get('piwik.settings');
+    $piwik_reporting_api_config = $this->configFactory->get('piwik_reporting_api.settings');
+    $site_id = $piwik_config->get('site_id');
+    $token_auth = $piwik_reporting_api_config->get('token_auth');
+
+    $sub_params = [
+      'module' => 'API',
+      'idSite' => $site_id,
+      'token_auth' => $token_auth,
+    ];
+
+    foreach ($items as $index => $item) {
+      if (!$entity = $this->getExpiredEntity($item)) {
+        continue;
+      }
+
+      $query_params = $this->getSubqueryParams($entity, $sub_params);
+      $query->setParameter('urls[' . $index . ']', http_build_query($query_params));
     }
-    // Only refresh the field if it has actually expired. It might have been
-    // updated already since it has been added to the processing queue.
-    if (!$this->fieldNeedsRefresh($event)) {
```

Comment:
This line is lost, let's keep it so we don't do unnecessary requests.

Reply:
Ok, I re-added it, though I do not think that this is a case in Joinup. But I cannot predict the future.